pkg/api/validation/validation.go

```diff
@@ -454,6 +454,14 @@ func validateHTTPGetAction(http *api.HTTPGetAction) errs.ValidationErrorList {
 	return allErrors
 }
 
+func validateTCPSocketAction(tcp *api.TCPSocketAction) errs.ValidationErrorList {
+	allErrors := errs.ValidationErrorList{}
+	if tcp.Port.IntVal == 0 {
```

Comment:
Do we want a comment about why this does not support named ports?  E.g. "Named ports are probably going to be deprecated, so don't handle that here."